packages/core/src/tracing/instrumentation/cloud/azure/blob.js

```diff
@@ -0,0 +1,153 @@
+/*
+ * (c) Copyright IBM Corp. 2023
+ */
+
+'use strict';
+
+const shimmer = require('../../../shimmer');
+
+const requireHook = require('../../../../util/requireHook');
+const tracingUtil = require('../../../tracingUtil');
+const constants = require('../../../constants');
+const cls = require('../../../cls');
+
+let isActive = false;
+
+exports.spanName = 'azstorage';
+
+exports.init = function init() {
+  requireHook.onModuleLoad('@azure/storage-blob', instrumentBlob);
+};
+
+function instrumentBlob(blob) {
+  instrumentBlockBlob(blob.BlockBlobClient);
+  instrumentBlobClient(blob.BlobClient);
```

Comment:
issue: Why do we not instrument [this.client.sendOperationRequest](https://github.com/search?q=repo%3AAzure%2Fazure-sdk-for-js%20this.client.sendOperationRequest&type=code)?

Reply:
Instrumenting BlockBlobClient and BlobClient give easy parsing for proper data required for span creation such as blobname, container name and operation. If using sendOperation, we need to parse it from the url and it doesn't seem neat. Also sendOperation is from @azure/core-http , and even though it is used in almost every operations in storage package, it doesn't act as common point of instrumentation when it comes to a major package like cosmosDB